internal/config/config.go

```diff
@@ -561,7 +561,8 @@ func poolsByNamespace(pools map[string]*Pool) map[string][]string {
 func poolsByServiceSelector(pools map[string]*Pool) []string {
 	var poolsByServiceSelector []string
 	for _, pool := range pools {
-		if pool.ServiceAllocations == nil || len(pool.ServiceAllocations.ServiceSelectors) == 0 {
+		if pool.ServiceAllocations == nil || len(pool.ServiceAllocations.Namespaces) > 0 ||
```

Comment:
is this change necessary? I think we should avoid the optimization of "checking both first, serviceselector only later" because it might bite us in the future. I guess it's cleaner if we have two separate sets, and each time we check for the whole compatiibility.

Reply:
ok, reverted it.